toucan_connectors/aircall/aircall_connector.py

```diff
@@ -1,24 +1,79 @@
+import asyncio
+import os
+from enum import Enum
 from typing import List, Optional, Tuple
 
 import pandas as pd
-import pyjq
+from aiohttp import ClientSession
 from pydantic import Field
 
-from toucan_connectors.common import FilterSchema, nosql_apply_parameters_to_query
+from toucan_connectors.common import get_loop
 from toucan_connectors.toucan_connector import ToucanConnector, ToucanDataSource
 
-PER_PAGE = 50
+from .constants import MAX_RUNS, PER_PAGE
+from .helpers import DICTIONARY_OF_FORMATTERS, build_df, build_empty_df
+
+BASE_ROUTE = f'https://proxy.bearer.sh/aircall_oauth'
+BEARER_API_KEY = os.environ.get('BEARER_API_KEY')
+
+
+async def fetch_page(
+    dataset: str,
+    data_list: List[dict],
+    session: ClientSession,
+    limit,
+    current_pass: int,
+    new_page=1,
+) -> List[dict]:
+    """
+    Fetches data from AirCall API
+
+    dependent on existence of other pages and call limit
+    """
+    endpoint = f'{BASE_ROUTE}/{dataset}?per_page={PER_PAGE}&page={new_page}'
+    data: dict = await fetch(endpoint, session)
+
+    data_list.append(data)
+
+    next_page_link = None
+    meta_data = data.get('meta')
+
+    if meta_data is not None:
+        next_page_link: Optional[str] = meta_data.get('next_page_link')
+
+    if limit > -1:
+        current_pass += 1
+
+        if next_page_link is not None and current_pass < limit:
+            next_page = meta_data.get('current_page') + 1
+            data_list = await fetch_page(
+                dataset, data_list, session, limit, current_pass, next_page
+            )
+    else:
+        if next_page_link is not None:
+            next_page = meta_data.get('current_page') + 1
```

Comment:
We usually use `.get()` on dictionaries when we're not sure the key is there. Here, you don't seem to handle that case so I would use `meta_data['current_page'] + 1` and let a `KeyError` propagate on errors.

(Same comment for the line 48)